envoy/http/query_params.h

```diff
@@ -15,6 +19,28 @@ namespace Utility {
 using QueryParams = std::map<std::string, std::string>;
 using QueryParamsVector = std::vector<std::pair<std::string, std::string>>;
 
+class QueryParamsMulti {
+private:
+  absl::btree_map<std::string, std::vector<std::string>> data;
+
+public:
+  void remove(absl::string_view key);
+  void add(absl::string_view key, absl::string_view value);
+  void overwrite(absl::string_view key, absl::string_view value);
+  std::string toString();
+  std::string replaceQueryString(const HeaderString& path);
+
+  // This function is only here to allow for sane unit testing, and should not be called outside of
+  // unit tests
+  absl::btree_map<std::string, std::vector<std::string>> copy_map_for_testing() {
+    return absl::btree_map<std::string, std::vector<std::string>>(data);
```

Comment:
hmm, we can just `return data;`
Besides, is this function going to be read only, if so, we can 
```
const absl::btree_map<std::string, std::vector<std::string>>&  data() {
    return data;
}
 ```

Generally returning reference raise the concern about lifetime, but here we are returning reference of class member, it is almost safe as people can not access if the class object is destroyed, unless there is a destroy function to clear this member while keeping the class object alive (which is not case here).